File: algorithms/python/powerofThree/powerofThree.py
#! /usr/bin/env python3
# -*- coding: utf-8 -*-

# Source: https://leetcode.com/problems/power-of-three/
# Author: Miao Zhang
# Date:   2021-02-02

class Solution:
    def isPowerOfThree(self, n: int) -> bool:
        return n > 0 and 1162261467 % n == 0

File: algorithms/cpp/powerofThree/powerofThree.cpp
// Source: https://leetcode.com/problems/power-of-three/
// Author: Miao Zhang
// Date:   2021-02-02

class Solution {
public:
    bool isPowerOfThree(int n) {
        if (n <= 0) return false;
        double a = log10(n) / log10(3);
        return a == floor(a);
    }
};
